datalad/support/network.py

```diff
@@ -812,6 +817,24 @@ def as_git_url(self):
         return "{}{}".format(consts.DATASETS_TOPURL, urlquote(self.path))
 
 
+class GitTransportRI(RI, RegexBasedURLMixin):
+    """RI for some other RI with git transport prefix"""
+
+    # TODO: check how crticial to "inherit" RI._FIELDS asking to provide path
+    _FIELDS = RI._FIELDS + (
+        'transport',
+        'RI',
+    )
+
+    # Due to poor design, `ri` argument already present in various
+    # places intermixed with **kwargs treatment. So we will use RI
+    # here instead of ri.
+    _REGEX = re.compile(r'(?P<transport>[^:/@]+)::(?P<RI>.*)$')
```

Comment:
I was curious what characters Git permits for transport.  As far as I can tell, it goes through `is_urlschemechar`, which carries this comment:

```
/*
 * The set of valid URL schemes, as per STD66 (RFC3986) is
 * '[A-Za-z][A-Za-z0-9+.-]*'. But use slightly looser check
 * of '[A-Za-z0-9][A-Za-z0-9+.-]*' because earlier version
 * of check used '[A-Za-z0-9]+' so not to break any remote
 * helpers.
 */
```

So, I think we could make this regexp stricter, though I doubt it matters much in practice.

Reply:
Thanks for digging it out, I looked only into a manpage -- I will adjust the regex to be `[A-Za-z0-9][A-Za-z0-9+.-]*`.

Reply:
```suggestion
    _REGEX = re.compile(r'(?P<transport>[A-Za-z0-9][A-Za-z0-9+.-]*)::(?P<RI>.*)$')
```